lib/repo_updater.rb

```diff
@@ -2,9 +2,20 @@
 
 require 'fileutils'
 
-# Update locally cached git repositories
+# Update locally cached git repositories, remove extraneous ones
 class RepoUpdater
   def self.update(repos:)
+    tmp_repos = Dir["#{Settings.work_dir}/*/*"].filter_map do |entry|
```

Comment:
I was thinking something like:

```suggestion
  def self.prune_removed_repos_from_cache!(repos)
    Dir["#{Settings.work_dir/*/*"].each do |cached_dir|
      next if repos.map(&:name).any? { |repo_name| cached_dir.ends_with?(repo_name) }
      
      FileUtils.rm_rf(cached_dir)
    end
  end
```

and then calling that method from `RepoUpdater.update`. Not sure it warrants a `puts` but maybe some might want to see that?